### PR TITLE
checks if hostname is .onion and change scheme

### PR DESCRIPTION
**Pull request by talvasconcelos**
I think this is the "issue" Super Testnet raised on the Telegram channel! 

Needs Tor testing, as I can't test it locally! @dni @motorina0 @callebtc ?

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -53,8 +54,11 @@ def from_row(cls, row: Row) -> "PayLink":
             data["max"] /= data["fiat_base_multiplier"]
         return cls(**data)
 
-    def lnurl(self, req: Request) -> str:
+    def lnurl(self, req: Request) -> str:
```
Comment (motorina0) [inline models.py:57]: - extra space, should format

File: models.py [hunk 2]
```diff
@@ -53,8 +54,11 @@ def from_row(cls, row: Row) -> "PayLink":
             data["max"] /= data["fiat_base_multiplier"]
         return cls(**data)
 
-    def lnurl(self, req: Request) -> str:
+    def lnurl(self, req: Request) -> str:        
         url = req.url_for("lnurlp.api_lnurl_response", link_id=self.id)
+        # Check if url is .onion and change to http
+        if ".onion" in url.hostname:
```
Comment (motorina0) [inline models.py:60]: - this rule is valid for `https://eat.some.onion.lnbits.com`
- should use `ends_with`

File: models.py [hunk 3]
```diff
@@ -53,8 +54,11 @@ def from_row(cls, row: Row) -> "PayLink":
...
+            url = url._replace(scheme="http")
```
Comment (motorina0) [inline models.py:61]: - `_replace` looks like a private method, is there a public one available?

**Comment by motorina0** [comment]
> Needs Tor testing, as I can't test it locally! @dni @motorina0 @callebtc ?

- can you please add testing steps

**Comment by callebtc** [comment]
I don't understand what this PR does, maybe you can explain. User enters a wrong URL and this PR corrects it? Why does the user enter a wrong URL in the first place?

**Comment by talvasconcelos** [comment]
> I don't understand what this PR does, maybe you can explain. User enters a wrong URL and this PR corrects it? Why does the user enter a wrong URL in the first place?

SuperTestnet reported an issue on telegram group about the QR from LNURLp, if lnbits is behind Tor, has the URL with https scheme, instead of http (i think it's the way FastAPI's `url_for` works)! This PR just check, before creating the link/callback if the url is a Tor (.onion) URL and replace `https` with `http`

**Review by dni — APPROVED** [review]

**Comment by motorina0** [comment]
@dni did you managed to test this PR?

**Pull request merged**